From 998911ec187fdbfce7a98a32340252948a369ffe Mon Sep 17 00:00:00 2001
From: Ariel D'Alessandro <ariel.dalessandro@collabora.com>
Date: Sat, 26 Apr 2025 19:46:32 -0300
Subject: [PATCH] Revert "[base] Introduce base::ToValueList<> and
 base::ToVector<>"

This reverts commit 2ffccb594d679cbbafc3e0ade85f4aa2555050a2.

[Revert only components/services/app_service/public/cpp/permission.cc]
Signed-off-by: Ariel D'Alessandro <ariel.dalessandro@collabora.com>
---
 components/services/app_service/public/cpp/permission.cc | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/components/services/app_service/public/cpp/permission.cc b/components/services/app_service/public/cpp/permission.cc
index ee82c80f96..cc5c816784 100644
--- a/components/services/app_service/public/cpp/permission.cc
+++ b/components/services/app_service/public/cpp/permission.cc
@@ -6,7 +6,6 @@
 
 #include <sstream>
 
-#include "base/containers/to_value_list.h"
 #include "base/strings/to_string.h"
 #include "third_party/abseil-cpp/absl/types/variant.h"
 
@@ -167,7 +166,11 @@ PermissionPtr ConvertDictToPermission(const base::Value::Dict& dict) {
 }
 
 base::Value::List ConvertPermissionsToList(const Permissions& permissions) {
-  return base::ToValueList(permissions, &ConvertPermissionToDict);
+  base::Value::List list;
+  for (const auto& permission : permissions) {
+    list.Append(ConvertPermissionToDict(permission));
+  }
+  return list;
 }
 
 Permissions ConvertListToPermissions(const base::Value::List* list) {
